Lib/test/test_random.py

```diff
@@ -923,6 +925,81 @@ def test_betavariate_return_zero(self, gammavariate_mock):
         gammavariate_mock.return_value = 0.0
         self.assertEqual(0.0, random.betavariate(2.71828, 3.14159))
 
+class TestRandomSubclassing(unittest.TestCase):
+    def test_random_subclass_with_kwargs(self):
+        # SF bug #1486663 -- this used to erroneously raise a TypeError
+        class Subclass(random.Random):
+            def __init__(self, newarg=None):
+                random.Random.__init__(self)
+        Subclass(newarg=1)
+
+    def test_overriding_random(self):
```

Comment:
This test uses implementation details. It is possible to test this behavior using only public API. Make `random()` and `getrandbits()` in subclasses logging their calls, and check what methods were used when call `randrange()`. This will make the test valid even if the implementation be changed.

Reply:
Good idea! I'll look into it.

Reply:
But maybe we need to keep a simple white-box test for random.Random._randbelow and a subclass that don't override random or getrandbits, because there is no way to test this as a black box. It is worth to mark this test as CPython-only.

Reply:
Sounds reasonable: I'm going to work on all the changes tomorrow, thanks.